src/main/java/net/dv8tion/jda/core/entities/impl/EmoteImpl.java

```diff
@@ -71,6 +71,7 @@ public EmoteImpl(long id,  JDA api)
         this.id = id;
         this.api = api;
         this.guild = null;
+        this.roles = null;
```

Comment:
This was only moved from line 58 into the constructor to allow a final field.
It would also be null for fake emotes without these changes.